data/mil2525d/utilities/military-features-utilities/template-gdb-toolbox/scripts/CreateTemplateGDB/AddFieldsFromSchema.py

```diff
@@ -0,0 +1,115 @@
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+# AddFieldsFromSchema.py
+# Description: Reads the field descriptions from a field schema CSV file and
+# adds those fields to the specified feature class.
+# Requirements: ArcGIS Desktop
+# ----------------------------------------------------------------------------------
+
+import arcpy, os, sys, traceback
+import csv as csv
+
+from arcpy import env
+from os import path
+
+def fieldTypeLookup(jmsmlFieldType):
+    # Converts a JMSML field type into an Esri field type
+    switcher = {
+                "Date": "DATE",
+                "Double": "DOUBLE",
+                "Integer": "LONG",
+                "SmallInteger": "SHORT",
+                "String": "TEXT",
+                }
+    
+    return switcher.get(jmsmlFieldType, "NONE")
+
+def castValue(valueType, originalValue):
+    # Cast input string value to a specified type
+    
+    if valueType == "DOUBLE":
+        castedValue = float(originalValue)
+    elif valueType == "LONG":
+        castedValue = int(originalValue)
+    elif valueType == "SHORT":
+        castedValue = int(originalValue)
+    else:
+        castedValue = originalValue
+        
+    return castedValue
+    
+def addFieldsFromSchema(schemasFolder, featureClass, schema):
+    
+    # Adds the fields defined in a field schema CSV file to the specified feature class
+    
+    try:
+        arcpy.AddMessage("Starting: AddFieldsFromSchema") 
+            
+        # Make sure the feature class exists
+        
+        if arcpy.Exists(featureClass):
+            
+            # Make sure the specfied field schema CSV file exists
+            
+            fieldSchemaFile = os.path.join(schemasFolder, "Fields_" + schema + ".csv")
+            if os.path.exists(fieldSchemaFile):
+                with open(fieldSchemaFile, 'r') as csvFile:
+                    reader = csv.reader(csvFile, dialect='excel')
+            
+                    # Skip the headers
+                    header = next(reader)
+                    
+                    # Read all the rows and add fields accordingly
+                    for line in reader:
+                        
+                        fieldType = fieldTypeLookup(line[1])
+                        
+                        if fieldType == "TEXT":
+                            length = int(line[2])
+                            
+                            
+                            arcpy.AddField_management(featureClass, \
+                                                      field_name=line[0], \
+                                                      field_type=fieldType, \
+                                                      field_length=length, \
+                                                      field_alias=line[3], \
+                                                      field_is_nullable=line[4])
```

Comment:
You are correct in your observation.  I purposefully left off the domain field until domain creation/updating is completed.  Setting a domain value when the domain doesn't yet exist is ignored by ArcGIS.  I didn't want to also take the risk when testing that it might behave strangely with the domain set prematurely, so I left it completely off.